types/src/transaction/authenticator.rs

```diff
@@ -785,6 +785,17 @@ impl MultiKeyAuthenticator {
     }
 
     pub fn verify<T: Serialize + CryptoHash>(&self, message: &T) -> Result<()> {
+        ensure!(
+            self.signatures_bitmap.last_set_bit().is_some(),
+            "There were no signatures set in the bitmap."
+        );
+
+        ensure!(
+            (self.signatures_bitmap.last_set_bit().unwrap() as usize) < self.public_keys.len(),
```

Comment:
nit: I think the two checks can be combined
```
last_set_bit().map_or(false, |idx| idx < self.public_keys.len())
```

we don't need to enforce the bitvec buckets, but from conservative perspective I prefer to enforce restrictions as much as possible

Reply:
If I combine the two checks, the error messages will not be as clear anymore. I'll leave them like this for now.